### PR TITLE
fix(administration): Update img query

### PR DESCRIPTION
**Pull request by cbadr-ext**
## Description

update a query

**Fixes** # (MON-37222)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

## Summary by CodeRabbit

- **Refactor**
	- Enhanced security and reliability of the image deletion process by using prepared statements and transaction handling.

### CODE REVIEW

File: centreon/www/include/options/media/images/DB-Func.php [hunk 1]
```diff
@@ -174,22 +174,37 @@ function deleteImg($imageId)
...
+            $deleteStatement->execute();
+
+            $deleteRelationQuery = "DELETE FROM view_img_dir_relation WHERE img_img_id = :imageId";
+            $deleteRelationStatement = $pearDB->prepare($deleteRelationQuery);
+            $deleteRelationStatement->bindParam(':imageId', $imageId, PDO::PARAM_INT);
+            $deleteRelationStatement->execute();
         }
-        $pearDB->query("DELETE FROM view_img WHERE img_id = $imageId");
-        $pearDB->query("DELETE FROM view_img_dir_relation WHERE img_img_id = $imageId");
+        $pearDB->commit();
```
Comment (nfragnet) [inline centreon/www/include/options/media/images/DB-Func.php:203]: I doubt it makes sense to loop on these two queries while the var passed to the placeholder ($imageId) is defined above the foreach and thus never changes.

You just loop on the exact same query with the exact same parameter.
Those two queries should probably move outside the foreach part.

(I know it was already the previous behaviour but let's still fix it)

File: centreon/www/include/options/media/images/DB-Func.php [hunk 2]
```diff
@@ -174,22 +174,37 @@ function deleteImg($imageId)
...
+    } catch (PDOException $e) {
+        $pearDB->rollBack();
+        echo "Error: " . $e->getMessage();
```
[the remark this answers is not in the document]
Comment (callapa) [reply centreon/www/include/options/media/images/DB-Func.php:205]: @cbadr-ext 

1. If you don't know, just ask us.
2. You must not present any technical errors to the user

It is not necessary to place the **SELECT** query in an SQL transaction.

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by callapa — COMMENTED** [review]
In general, we prefer to use the BindValue method rather than bindParam, which doesn't work in the same way.

**Comment by cbadr-ext** [comment]
> In general, we prefer to use the BindValue method rather than bindParam, which doesn't work in the same way.

> done i fix it

File: centreon/www/include/options/media/images/DB-Func.php [hunk 3]
```diff
@@ -175,21 +175,34 @@ function deleteImg($imageId)
...
+        $deleteStatement = $pearDB->prepare('DELETE FROM view_img WHERE img_id = :imageId');
+        $deleteStatement->bindValue(':imageId', $imageId, PDO::PARAM_INT);
+        $deleteStatement->execute();
+
+        $deleteRelationStatement = $pearDB->prepare('DELETE FROM view_img_dir_relation WHERE img_img_id = :imageId');
+        $deleteRelationStatement->bindValue(':imageId', $imageId, PDO::PARAM_INT);
+        $deleteRelationStatement->execute();
+        $pearDB->commit();
+    } catch (PDOException $e) {
+        $pearDB->rollBack();
```
Comment (callapa) [inline centreon/www/include/options/media/images/DB-Func.php:203]: As you catch exceptions, if an error occurs, the problem will be hidden.
In this case, throw the exception.
```suggestion
        $pearDB->rollBack();
        throw $e;
```

**Review by callapa — CHANGES_REQUESTED** [review]

File: centreon/www/include/options/media/images/DB-Func.php [hunk 4]
```diff
@@ -338,26 +351,24 @@ function deleteDirectory($directoryId)
...
-     * Purge images of the directory
-     */
-    $dbResult = $pearDB->query(
-        "SELECT img_img_id "
-        . "FROM view_img_dir_relation "
-        . "WHERE dir_dir_parent_id = $directoryId"
-    );
-    while ($img = $dbResult->fetch()) {
+
+// Purge images of the directory
```
Comment (callapa) [inline centreon/www/include/options/media/images/DB-Func.php:356]:
```suggestion
    // Purge images of the directory
```

File: centreon/www/include/options/media/images/DB-Func.php [hunk 5]
```diff
@@ -366,8 +377,12 @@ function deleteDirectory($directoryId)
         }
     }
     rmdir($mediadir . $safeDirAlias);
+
     if (!is_dir($mediadir . $safeDirAlias)) {
-        $dbResult = $pearDB->query("DELETE FROM view_img_dir WHERE dir_id = $directoryId");
+        $statement3 = $pearDB->prepare("DELETE FROM view_img_dir WHERE dir_id = :directoryId");
+        $statement3->bindValue(':directoryId', $directoryId, PDO::PARAM_INT);
+        $statement3->execute();
+        $statement3->closeCursor();
```
Comment (callapa) [inline centreon/www/include/options/media/images/DB-Func.php:385]: Nor necessary
```suggestion
```

File: centreon/www/include/options/media/images/DB-Func.php [hunk 6]
```diff
@@ -174,22 +174,37 @@ function deleteImg($imageId)
...
         $fullpath = $mediadir . basename($imagePath["dir_alias"]) . "/" . basename($imagePath["img_path"]);
         if (is_file($fullpath)) {
             unlink($fullpath);
         }
-        $pearDB->query("DELETE FROM view_img WHERE img_id = $imageId");
-        $pearDB->query("DELETE FROM view_img_dir_relation WHERE img_img_id = $imageId");
     }
-    $dbResult->closeCursor();
+    try {
+        $pearDB->beginTransaction();
```
Comment (nfragnet) [inline centreon/www/include/options/media/images/DB-Func.php:195]:
```suggestion
    $pearDB->beginTransaction();
    try {
```

File: centreon/www/include/options/media/images/DB-Func.php [hunk 7]
```diff
@@ -174,22 +174,38 @@ function deleteImg($imageId)
...
         $fullpath = $mediadir . basename($imagePath["dir_alias"]) . "/" . basename($imagePath["img_path"]);
         if (is_file($fullpath)) {
             unlink($fullpath);
         }
-        $pearDB->query("DELETE FROM view_img WHERE img_id = $imageId");
-        $pearDB->query("DELETE FROM view_img_dir_relation WHERE img_img_id = $imageId");
     }
-    $dbResult->closeCursor();
+
+        $pearDB->beginTransaction();
```
Comment (nfragnet) [inline centreon/www/include/options/media/images/DB-Func.php:195]:
```suggestion
    $pearDB->beginTransaction();
```

**Review by nfragnet — APPROVED** [review]

**Review by nfragnet — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**